aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -1138,6 +1139,76 @@ Tensor view_nested(const Tensor& self, IntArrayRef proposed_shape) {
       "Use .reshape(...) instead.");
   return create_nested_view_tensor(self, sizemat_reshaped, stridemat_reshaped, std::vector<int64_t>(self_ptr->get_offsets()));
 }
+  /**
+   * Create a buffer tensor that is a view of self
+   *
+   * This serves as the boundary between nested and non nested tensor
+   * view conversions
+   *
+   * @return Returns a new non nested tensor that
+   * aliases the same storage as self
+   */
+Tensor values_nested(const Tensor& self) {
+  TORCH_INTERNAL_ASSERT(self.is_nested(), "Can only create a buffer from Nested Tensor");
+  auto* nt_self = get_nested_tensor_impl(self);
+  return nt_self->get_buffer();
+}
+
+/**
+ * Create a nested tensor that is a view of a buffer
+ *
+ * This serves as the boundary between non nested tensor and nested
+ * view conversions
+ *
+ * @return Returns a nested tensor that
+ * aliases the same storage as buffer
+ */
+Tensor _nested_view_from_buffer(
+    const Tensor& buffer,
+    const Tensor& nested_size_tensor,
+    const Tensor& nested_stride_tensor,
+    IntArrayRef offsets) {
+  TORCH_INTERNAL_ASSERT(
+      !buffer.is_nested(),
+      "Can only a create Nested Tensor from a normal tensor buffer");
+  TORCH_INTERNAL_ASSERT(buffer.dim() == 1, "The input buffer must be flat");
+  TORCH_INTERNAL_ASSERT(nested_size_tensor.dim() == 2, "Expected the nested size tensor to be two dimensional.");
+  uint64_t num_elements_nested_size = at::prod(nested_size_tensor, 1).sum().item<int64_t>();
+  uint64_t buffer_storage_size = buffer.storage().nbytes()/buffer.dtype().itemsize();
+  TORCH_INTERNAL_ASSERT(
+      buffer_storage_size == num_elements_nested_size,
+      "The number of elements in the buffer must equal the nested tensor size but buffer size: ",
+      buffer_storage_size,
+      " and nested tensor size: ",
+      num_elements_nested_size,
+      ".");
+
+  TORCH_INTERNAL_ASSERT(nested_stride_tensor.dim() == 2, "Expected the nested stride tensor to be two dimensional.");
+  TORCH_INTERNAL_ASSERT(nested_size_tensor.size(0) == nested_stride_tensor.size(0), "Expected the first dimension of nested size and nested stride tensor to be equal.");
+  TORCH_INTERNAL_ASSERT(nested_stride_tensor.size(0) == (int64_t)offsets.size(), "Expected the first dimension of nested stride tensor to equal the length of offsets.");
+  return at::detail::make_tensor<NestedTensorImpl>(
+    c10::TensorImpl::VIEW,
+    buffer,
+    nested_size_tensor,
+    nested_stride_tensor,
+    std::vector<int64_t>(offsets.begin(), offsets.end()));
+}
+
+Tensor _nested_from_buffer(
+    const Tensor& buffer,
+    const Tensor& nested_size_tensor,
+    const Tensor& nested_stride_tensor,
+    IntArrayRef offsets) {
+  TORCH_INTERNAL_ASSERT(
+      !buffer.is_nested(),
+      "Can only a create Nested Tensor from a normal tensor buffer");
+  TORCH_CHECK(buffer.dim() == 1, "The input buffer must be flat");
```

Comment:
These checks are redundant, since they're also called in `_nested_view_from_buffer`, right?

Is the main purpose of `_nested_from_buffer` just that it's a thin wrapper around `_nested_view_from_buffer` that calls `.contiguous()` first to handle broadcasts, and is needed for the backward formula to `values()` (so it's not directly user-facing)?

Maybe we should just add it to `FunctionsManual.cpp` then, and not bother making it an aten / dispatcher operator (which implies that it should be overrideable, has more BC implications once we make it public, etc).

Is there any reason that `_nested_from_buf

Reply:
`_nested_from_buffer` is also even more annoying as an ATen op because it's another "might return an alias" op (since `.contiguous()` might give you a fresh tensor).

Reply:
Spoke with Brian offline. Since from_buffer is only being used as a backward formula. Going to remove the maybe_view from_buffer version do the contiguous and view call in derivatives.yaml.